### PR TITLE
Change return value of `Interval::merge` to always return a value

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

The `Interval::merge` operation currently returns an iterator, which always has a single value. Instead, we can return the value directly.

## 🔍 What does this change?

- Change the return value of `Interval::merge` from `impl ExactSizeIterator<Item = Self>` to `Self`

## 🛡 What tests cover this?

- The Rust compiler
- The unit tests of `Interval` were adjusted to use only a single value instead of an iterator

### CODE REVIEW

File: packages/graph/hash_graph/lib/interval-ops/src/interval.rs [hunk 1]
```diff
@@ -183,7 +183,7 @@ pub trait Interval<T>: Sized {
         // Range A:   [-----] | [-----]   | [-----]         |         [-----] | [---------]
         // Range B: [-----]   |   [-----] |         [-----] | [-----]         |   [-----]
         // Result:  [-------] | [-------] | [-------------] | [-------------] | [---------]
```
Comment (Alfred-Mountfield) [inline packages/graph/hash_graph/lib/interval-ops/src/interval.rs:185]: This confused me for quite a while, can you add a heading above each column to explain they're examples or something (e.g. `Example 1 | Example 2`, or maybe just a top level heading saying "Each column constitutes an example"

File: packages/graph/hash_graph/lib/interval-ops/src/interval.rs [hunk 2]
```diff
@@ -554,15 +553,15 @@ mod tests {
...
             [excluded_excluded(0, 5)],
         );
         test(
             included_included(5, 15),
             excluded_included(0, 10),
             [included_included(5, 10)],
             [excluded_included(0, 15)],
-            [excluded_included(0, 15)],
+            excluded_included(0, 15),
             [excluded_included(10, 15)],
```
Comment (Alfred-Mountfield) [inline packages/graph/hash_graph/lib/interval-ops/src/interval.rs:565]: Do we want to consider making a struct for these? I appreciate these are tests but it'd be too easy to accidentally swap one of these around.

**Review by Alfred-Mountfield — APPROVED** [review]
Some thoughts, but fine to merge without

File: packages/graph/hash_graph/lib/interval-ops/src/interval.rs (hunk 2 above)
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/interval-ops/src/interval.rs:565]: This is just a shorthand for
```rust
IntervalBounds::<u32>::from_bounds(Bound::Excluded(0), Bound::Included(15))
```
to make it more readable

Comment (Alfred-Mountfield) [reply packages/graph/hash_graph/lib/interval-ops/src/interval.rs:565]: Sorry I mean encapsulating the parameters to the actual `test` function

Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/interval-ops/src/interval.rs:565]: I cannot really follow. Could you give an example?

Comment (Alfred-Mountfield) [reply packages/graph/hash_graph/lib/interval-ops/src/interval.rs:565]:
```Rust
test(IntervalTestParameters {
    lhs: included_included(5, 15),
    rhs: excluded_included(0, 10),
    intersection: [included_included(5, 10)],
    union: [excluded_included(0, 15)],
    merge: excluded_included(0, 15),
    difference:[excluded_included(10, 15)]
}

Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/interval-ops/src/interval.rs:565]: Oh, yes, that makes sense. I will adjust this as part of #1838

File: packages/graph/hash_graph/lib/interval-ops/src/interval.rs (hunk 1 above)
Comment (Alfred-Mountfield) [reply packages/graph/hash_graph/lib/interval-ops/src/interval.rs:185]: Thanks, I'll respond there

**Pull request merged**